addons/sale/models/sale_order.py

```diff
@@ -33,21 +33,26 @@ def _get_default_require_signature(self):
     def _get_default_require_payment(self):
         return self.env.company.portal_confirmation_pay
 
-    @api.depends('order_line.price_total')
+    @api.depends('order_line.price_subtotal', 'order_line.price_tax', 'order_line.price_total')
     def _amount_all(self):
         """
         Compute the total amounts of the SO.
         """
         for order in self:
-            amount_untaxed = amount_tax = 0.0
-            for line in order.order_line:
-                amount_untaxed += line.price_subtotal
-                amount_tax += line.price_tax
-            order.update({
-                'amount_untaxed': amount_untaxed,
-                'amount_tax': amount_tax,
-                'amount_total': amount_untaxed + amount_tax,
-            })
+            order_lines = order.order_line.filtered(lambda x: not x.display_type)
+            tax_results = self.env['account.tax']._compute_taxes(
+                [self.env['account.tax']._prepare_base_line(
+                    x,
+                    currency=order.currency_id,
+                    taxes=x.tax_id,
+                    quantity=x.product_uom_qty,
+                ) for x in order_lines],
```

Comment:
Since the same code is duplicated thrice, wouldn't it make sense to factorize it ?
```suggestion
                order.order_line._prepare_tax_calculation_inputs(),
```

```python
def _prepare_tax_calculation_inputs(self):
    return [self.env['account.tax']._prepare_base_line(
        line,
        currency=line.currency_id,
        taxes=line.tax_id,
        quantity=line.product_uom_qty,
    ) for line in self if not line.display_type]
```